src/Common/src/TypeSystem/Common/ThrowHelper.cs

```diff
@@ -54,6 +54,12 @@ public static void ThrowBadImageFormatException()
             throw new TypeSystemException.BadImageFormatException();
         }
 
+        [System.Diagnostics.DebuggerHidden]
+        public static void ThrowRequiresRuntimeJitException(string reason)
```

Comment:
Making this a `TypeSystemException` with an `ExceptionStringId` is unnecessary. These have a lot of ceremony around the exception messages because they need to support scenarios where we throw them during compilation (where they need to be presented with a message that matches the locale of the compiler because they show up in warnings), and then we can also generate throwing stubs from them that throw at runtime, using the locale of the class library.

You can just make your own exception type local to ILCompiler.ReadyToRun.

Reply:
(You missed updating the places that deal with the ExceptionStringID anyway.)

Reply:
Fixed in 4th commit.